test/utils/read-rows-resumption.ts

```diff
@@ -0,0 +1,182 @@
+import * as assert from 'assert';
+import {GoogleError} from 'google-gax';
+
+describe('Bigtable/Utils/ReadrowsResumptionStrategy', () => {
+  const tableName = 'fake-table-name';
+  [
+    {
+      name: 'should generate the right resumption request with no options each time',
+      expectedResumeRequest: {
+        rows: {
+          rowKeys: [],
+          rowRanges: [{}],
+        },
+        tableName,
+      },
+      options: {},
+    },
+    {
+      name: 'should generate the right resumption requests with a last row key',
+      expectedResumeRequest: {
+        rows: {
+          rowKeys: ['c'].map(key => Buffer.from(key)),
+          rowRanges: [],
+        },
+        tableName,
+      },
+      options: {
+        keys: ['a', 'b', 'c'],
+      },
+      lastRowKey: 'b',
+    },
+    {
+      name: 'should generate the right resumption request with the lastrow key in a row range',
+      expectedResumeRequest: {
+        rows: {
+          rowKeys: [],
+          rowRanges: [
+            {startKeyOpen: Buffer.from('b'), endKeyClosed: Buffer.from('c')},
+            {startKeyClosed: Buffer.from('e'), endKeyClosed: Buffer.from('g')},
+          ],
+        },
+        tableName,
+      },
+      options: {
+        ranges: [
+          {start: 'a', end: 'c'},
+          {start: 'e', end: 'g'},
+        ],
+      },
+      lastRowKey: 'b',
+    },
+    {
+      name: 'should generate the right resumption request with the lastrow key at the end of a row range',
+      expectedResumeRequest: {
+        rows: {
+          rowKeys: [],
+          rowRanges: [
+            {startKeyClosed: Buffer.from('e'), endKeyClosed: Buffer.from('g')},
+          ],
+        },
+        tableName,
+      },
+      options: {
+        ranges: [
+          {start: 'a', end: 'c'},
+          {start: 'e', end: 'g'},
+        ],
+      },
+      lastRowKey: 'c',
+    },
+    {
+      name: 'should generate the right resumption request with start and end',
+      expectedResumeRequest: {
+        rows: {
+          rowKeys: [],
+          rowRanges: [
+            {
+              startKeyOpen: Buffer.from('d'),
+              endKeyClosed: Buffer.from('m'),
+            },
+          ],
+        },
+        tableName,
+      },
+      options: {
+        start: 'b',
+        end: 'm',
+      },
+      lastRowKey: 'd',
+    },
+    {
+      name: 'should generate the right resumption request with prefixes',
+      expectedResumeRequest: {
+        rows: {
+          rowKeys: [],
+          rowRanges: [
+            {
+              startKeyClosed: Buffer.from('f'),
+              endKeyOpen: Buffer.from('g'),
+            },
+            {
+              startKeyClosed: Buffer.from('h'),
+              endKeyOpen: Buffer.from('i'),
+            },
+          ],
+        },
+        tableName,
+      },
+      options: {
+        prefixes: ['d', 'f', 'h'],
+      },
+      lastRowKey: 'e',
+    },
```

Comment:
I'm not really seeing any test cases where both keys and ranges need to be modified at the same time

Reply:
It might also be good to add some test cases where nothing happens (lastRowKey is < the queries)

Reply:
Added these test cases.